GitCommands/Git/GitModule.cs

```diff
@@ -970,7 +970,10 @@ public GitRevision GetRevision(ObjectId? objectId = null, bool shortFormat = fal
                 objectId
             };
 
-            var revInfo = _gitExecutable.GetOutput(args, cache: GitCommandCache, outputEncoding: LosslessEncoding);
+            // cache output only if revision is specified
+            CommandCache? cache = objectId is null ? null : GitCommandCache;
```

Comment:
Related: Cache would not be valid if called with an artificial commit, but the command will then fail anyway, so no check needed here...
Do not suggest a check in the method